types/react-native/index.d.ts

```diff
@@ -8239,6 +8240,11 @@ export interface StatusBarProps extends StatusBarPropsIOS, StatusBarPropsAndroid
      */
     animated?: boolean;
 
+    /**
+     * Sets the color of the status bar text.
+     */
+    barStyle?: StatusBarStyle;
```

Comment:
If we are precise, `barStyle` could also be `null` (see https://flow.org/en/docs/types/maybe/). The same would apply to `showHideTransition`, as both are a maybe type in the original source code :)

Reply:
I strongly believe this is Flow's opinion. While it might "work" in this case, it certainly does not make sense in other cases.

We might see this behaviour in whole type def here :)

Also, if I ever want to revert back to system default (let's omit presence of 'default' value here) and I am not sure about default value, I would like to have one way to do so.

```tsx
<StatusBar barStyle={undefined}  />

// vs

<StatusBar barStyle={null}  />
```

What do you think?

Reply:
Hm, I don’t fully understand your point. As both options are possible, all of them would set it back to default (`undefined` as well as `null`, or not setting the property at all, making it `undefined` as well)? So what’s speaking against allowing all possible values as defined by the original typings?

Reply:
added `null` as allowed value